src/web/views.py

```diff
@@ -39,35 +40,36 @@ def __init__(self, **kwargs: Any) -> None:
         super().__init__(**kwargs)
 
 
-class DetailEvent(DetailView):
+class DetailEvent(HtmxViewMixin, DetailView):
     model = Event
 
-
-def list_tech_groups(request: HttpRequest) -> HttpResponse:
-    groups = TechGroup.objects.all()
-    return render(request, "web/list_tech_groups.html", { "groups": groups })
-
-
-def get_tech_group(request: HttpRequest, pk: int) -> HttpResponse:
-    group = TechGroup.objects.get(pk=pk)
-    return render(request, "web/get_tech_group.html", { "group": group })
+    def get(self, request, *args, **kwargs):
+        if self.is_htmx():
+            self.template_name = "web/partials/detail_event.htm"
+        return super().get(request, *args, **kwargs)
 
 
-def get_event(request: HttpRequest, pk: int) -> HttpResponse:
-    event = Event.objects.get(pk=pk)
-    return render(request, "web/get_tech_group.html", { "event": event })
+class DetailTechGroup(HtmxViewMixin, DetailView):
+    model = TechGroup
 
+    def get(self, request, *args, **kwargs):
+        if self.is_htmx():
+            self.template_name = "web/partials/detail_tech_group.htm"
+        return super().get(request, *args, **kwargs)
 
-class GetTechGroups(GenericHtmxView):
-    """Get a list of enabled TechGroups and render a partial to use on the sidebar navigation"""
-    template_name = "web/partials/sidebar_items.htm"
-    queryset = TechGroup.objects.filter(enabled=True)
 
+def list_tech_groups(request: HttpRequest) -> HttpResponse:
+    groups = TechGroup.objects.all()
+    return render(request, "web/list_tech_groups.html", { "groups": groups })
 
-class GetEvents(GenericHtmxView):
-    """Get a list of upcoming Events and render a partial to use on the sidebar navigation"""
-    template_name = "web/partials/sidebar_items.htm"
 
-    def __init__(self, **kwargs):
-        super().__init__(**kwargs)
-        self.queryset = Event.objects.filter(date_time__gte=timezone.now())
+class BuildSidebar(BuildModelSidebarNav):
+    """Get a list of upcoming Events and enabled TechGroups and render a partial to use on the sidebar navigation"""
+    menu_item_list = [
+        {"queryset": Event.objects.filter(date_time__gte=timezone.now()),
```

Comment:
I think `timezone.now()` will only be evaluated once since it's in a class variable, instead of every time an instance of this class is created. Probably not what we want.

Reply:
This is actually handled by by the parent class. If you test locally, create an event in the future, verify it's included in the menu and then update the event date in the admin and the page refresh; the event should no longer appear in the menu.

Reply:
ah right on! Sorry, should have tested this locally before commenting.